core/base/topologicalSimplification/TopologicalSimplification.h

```diff
@@ -168,149 +120,38 @@ namespace ttk {
                 const SimplexId *const inputOffsets,
                 SimplexId *const offsets,
                 const SimplexId constraintNumber,
-                const triangulationType &triangulation) const;
+                const bool addPerturbation,
+                const triangulationType &triangulation);
+
+    inline void setBackend(const std::string &arg) {
+      if(arg == "lts")
+        backend_ = TOPO_LTS;
+      else if(arg == "legacy")
+        backend_ = TOPO_LEGACY;
+      else
+        this->printMsg("Error : topological simplification should be 'lts' or "
+                       "'legacy', but got: "
+                       + arg);
+    }
 
     inline int preconditionTriangulation(AbstractTriangulation *triangulation) {
       if(triangulation) {
-        vertexNumber_ = triangulation->getNumberOfVertices();
         triangulation->preconditionVertexNeighbors();
+        if(backend_ == TOPO_LEGACY)
+          legacyObject_.setNumberOfVertices(
+            triangulation->getNumberOfVertices());
       }
       return 0;
     }
 
-    inline void setConsiderIdentifierAsBlackList(bool onOff) {
-      considerIdentifierAsBlackList_ = onOff;
-    }
-
-    inline void setAddPerturbation(bool onOff) {
-      addPerturbation_ = onOff;
-    }
-
   protected:
-    SimplexId vertexNumber_{};
-    bool considerIdentifierAsBlackList_{false};
-    bool addPerturbation_{false};
+    enum TOPO_BACKEND { TOPO_LTS, TOPO_LEGACY };
```

Comment:
SIMPLIFICATION_BACKEND

Reply:
maybe use the PersistenceDiagram style instead (enum declared within the class, simpler)